src/engine/engine.rs

```diff
@@ -20,13 +20,23 @@ pub trait HasName: Debug {
     fn name(&self) -> &str;
 }
 
-pub trait Dev: Debug {}
-
 pub trait Filesystem: HasName + HasUuid {
     /// path of the device node
     fn devnode(&self) -> PathBuf;
 }
 
+pub trait BlockDev: HasUuid {
+    /// path of the device node
+    fn devnode(&self) -> PathBuf;
+
+    fn user_id(&self) -> &Option<String>;
+    fn set_user_id(&mut self, location: Option<&str>) -> EngineResult<()>;
+    fn hardware_id(&self) -> &Option<String>;
+    fn initialization_time(&self) -> String;
```

Comment:
I'ld rather this be a TimeStamp...let D-Bus API layer convert it to its preferred format.

Reply:
You mean a `DateTime<Utc>`? Makes sense.